### DISCUSSION TITLE
Accessing additional Sentinel 2 products?

### DISCUSSION THREAD

**Discussion by JackDunnNZ** [cogeotiff/rio-tiler-pds#73] [Q&A]
I am trying to access the SCL product from Sentinel 2, but it seems like it is not listed in the list of available bands:

```python
with S2COGReader("S2A_28QED_20230514_0_L2A") as sentinel:
    img = sentinel.preview("SCL")
```

gives 

```
InvalidBandName: SCL is not valid.
Valid bands: ('B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B8A', 'B09', 'B11', 'B12')
```

However if I modify the bands to add it as a valid band, everything seems to work fine:

```
with S2COGReader("S2A_28QED_20230514_0_L2A") as sentinel:
    sentinel.bands += ("SCL",)
    img = sentinel.preview("SCL")
imshow(img.data_as_image())
```

![image](https://github.com/cogeotiff/rio-tiler-pds/assets/4717044/c45d21d6-663b-4859-b6c4-146e065670bd)

I wanted to check if there was a particular reason these products weren't available (e.g. a correctness issue and thus I should definitely avoid doing this) or if it was perhaps an oversight?

**Comment by vincentsarago** [comment]
Interesting the product are commented out in https://github.com/cogeotiff/rio-tiler-pds/blob/a1e9677baeec3b2b63813499811aac3652de2f50/rio_tiler_pds/sentinel/aws/sentinel2.py#L146-L148

I can't remember why we did that 

In the docs there is a note:

> Note: AOT, SCL, WVP STAC assets are not supported.

https://cogeotiff.github.io/rio-tiler-pds/usage/sentinel/#cog-only-l2a-available-for-now

I guess it's mainly because the `Readers` are supposed to work with `Bands` which would explain why we removed the `Products` like AOT, SCL and WVP

**Comment by JackDunnNZ** [comment]
Yeah, I went looking through the history to try and see if there was a reason for those to be commented out, and didn't find anything. As long as I'm not doing anything obviously unsafe by getting the products in the same way, that is good for me. Thanks!
